tutorials/FITS-header/FITS-header.ipynb

```diff
@@ -18,187 +20,288 @@
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "metadata": {
+    "id": "YKrZaxbRmv2a"
+   },
    "outputs": [],
    "source": [
-    "fits.writeto('output_file.fits', data, header, overwrite=True)"
+    "# But hdu_number = 0 is the PRIMARY HDU. How many HDUs are in this file?\n",
+    "fits_inf = fits.open(\"input_file.fits\")\n",
+    "fits_inf.info() \n",
+    "fits_inf[0].header"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "id": "z_ddsyRUoVu8"
+   },
    "source": [
-    "That's it; you're done!"
+    "**Thought Question:** *How many HDUS are in this file? Were you right about the number of cards in the primary HDU?*"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "id": "AKTWFAyeopR4"
+   },
    "source": [
-    "Two common and more complicated cases are worth mentioning (but if your needs \n",
-    "are much more complex, you should consult the full documentation http://docs.astropy.org/en/stable/io/fits/). \n",
-    "\n",
-    "The first complication is that the FITS file you're examining and \n",
-    "editing might have multiple HDU's (extensions), in which case you can \n",
-    "specify the extension like this:"
+    "**Thought Question:** *What will happen if you type* `fits_inf[2].header`*?*"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "UFYUH450mv2b"
+   },
+   "source": [
+    "Now that we know how to read out FITS files, let's change the header. The header is always changed via it’s keyword. The keywords are the 8 letter word to the left when you extract the header info. Lets change change the header to give the correct OBJECT:"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "metadata": {
+    "id": "LOP18OR0mv2b"
+   },
    "outputs": [],
    "source": [
-    "data1, header1 = fits.getdata(\"input_file.fits\", ext=1, header=True)"
+    "header['OBJECT'] = \"M31\""
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {},
+   "metadata": {
+    "id": "tHfDDzD9o8Ty"
+   },
    "source": [
-    "This will get you the data and header associated with the `index=1` extension \n",
-    "in the FITS file. Without specifying a number, `getdata()` will get the \n",
-    "0th extension (equivalent to saying `ext=0`)."
+    "The keyword OBJECT was in the primary HDU. \n",
+    "\n",
+    "<div class=\"alert alert-info\">\n",
+    "\n",
+    "NOTE: If we wanted to  change a header in another HDU, we could do it by executing the following code. \n",
+    "\n",
+    "```\n",
+    "hdr = hdul[1].header\n",
+    "hdr[‘EXAMPLE’] = ‘example :)’\n",
+    "```\n",
+    "\n",
+    "This won't actually run becasue we dont have a EXAMPLE keyword in HDU 1, but this could be useful in the future.\n",
```

Comment:
I think this should run. If a keyword doesn't already exist in the header then it is simply added. I'd suggest making this a code cell to demonstrate how to add keywords.